lib/exercises.js

```diff
@@ -1,17 +1,68 @@
 // This method will return an array of arrays.
 // Each subarray will have strings which are anagrams of each other
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: O(m) m is all of the characters in all of the words
+// Space Complexity: O(n+n) -> O(n)
 function groupedAnagrams(strings) {
-    throw new Error("Method hasn't been implemented yet!");
+    let freqMap = new Map();
+
+    let result = [];
+    for (let word of strings) {
+        let count = new Array(26).fill(0);
+        for (let char of word) {
+            count[char.charCodeAt(0) - 97] += 1;
+        }
+
+        if (freqMap.get(count.toString())) {
+            freqMap.get(count.toString()).push(word);
+        } else {
+            freqMap.set(count.toString(), [word]);
+        }
+    }
+    
+    for (let list of freqMap.values()) {
+        result.push(list);
+    }
+    return result;
 }
 
 // This method will return the k most common elements
 // in the case of a tie it will select the first occuring element.
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: O(n)
+// Space Complexity: O(n+n+n) -> O(n)
 function topKFrequentElements(list, k) {
-    throw new Error("Method hasn't been implemented yet!");
+
+    let counts = new Map();
+    let buckets = [];
+    for (let i = 0; i <= list.length; i++)
+        buckets.push([]);
+    
+    // count frequent of the elements
+    for (let num of list) {
+        if (counts.has(num)) {
+            counts.set(num, counts.get(num) + 1);
+        } else {
+            counts.set(num, 1);
+        }
+    } 
+    // put them into buckets by frequent
+    for (let [key, value] of counts) {
+        buckets[value].push(key);
+    }
+
+    console.log(buckets);
+    // fetch the larget frequest bucket first, until reach k
+    let ans = [];
+    for (let i = buckets.length - 1; i > 0 && ans.length < k; i--) {
+        if (buckets[i] !== null) ans.push(...buckets[i]);
```

Comment:
There's a slight bug here that I'm surprised didn't break the tests. To initialize `buckets` you added an empty array `[]` to each bucket. Those empty buckets are getting added to the `ans` array too because empty arrays are not `!==` to `null`. A better check would be `buckets[i].length !== 0`. (I guess you could `&&` that with a check for `null`, but with your current code, none of the elements in `buckets` will be `null`.)

Reply:
Note that this bug isn't so major that I'm moving down to Yellow.

Reply:
I appreciate your feedback, but a small note about your phrasing "Note that this bug isn't so major that I'm moving down to Yellow." and "this shows adequate demonstration of the material so this is good enough for a Green" are not the most constructive of comments.